<?xml version="1.0"?>

<!-- map.launch -->
<launch>
    <!-- Waypoints -->
    <arg name="simulation" default="false"/>
    <include file="$(find igvc_navigation)/launch/waypoint_source.launch"
             unless="$(arg simulation)"/>

    <!-- Planner -->
    <node name="path_planner" pkg="igvc_navigation" type="pathplanner" output="screen" >
        <param name="goal_threshold" type="double" value="0.5"/>
        <param name="c_space" type="double" value="0.5"/>
        <param name="probability_threshold" type="double" value="0.7"/>
        <param name="point_turns_enabled" type="bool" value="false"/>
        <param name="reverse_enabled" type="bool" value="false"/>
        <param name="max_jump_size" type="double" value="10.0"/>
        <param name="theta_filter" type="double" value="1.5"/>
        <param name="max_theta_change" type="double" value="0.8"/>
        <param name="theta_change_window" type="double" value="3.0"/>
        <!-- Max distance to a goal node -->
        <param name="maximum_distance" type="double" value="100.0"/>
    </node>

    <!-- Follower -->
    <node name="path_follower" pkg="igvc_navigation" type="path_follower" output="screen" >
        <!-- Maxmimum velocity allowed before stopping motors -->
        <param name="maximum_vel" type="double" value="2.0"/>
        <param name="axle_length" type="double" value="0.52"/>
        <!-- Distance along the path to generate smooth control law for  -->
        <param name="lookahead_dist" type="double" value="1.3"/>
        <!-- granularity with which to visualize planned trajectories -->
        <param name="granularity" type="int" value="100"/>
        <!-- How far away robot should be from the waypoint before it stops -->
        <param name="stop_dist" type="double" value="0.90"/>
        <!-- Smooth Control Parameters -->
        <param name="target_v" type="double" value="0.4"/>
        <param name="k1" type="double" value="1.8"/>
        <param name="k2" type="double" value="5"/>
        <param name="debug" type="bool" value="true"/>
    </node>

</launch>
